main/lsp/notifications/initialized.cc

```diff
@@ -13,8 +13,8 @@ void InitializedTask::preprocess(LSPPreprocessor &preprocessor) {
 }
 
 void InitializedTask::index(LSPIndexer &indexer) {
-    // Hacky: We need to use the indexer, but with the WorkerPool from runSpecial. This is the only task to have this
-    // special requirement.
+    // Hacky: We need to use the indexer, but with the WorkerPool from runSpecial. This + SorbetWorkspaceEdit are the
+    // only tasks to have this special requirement.
```

Comment:
Not necessarily in this change, but given that this hack has grown, is there a restructuring that would make it easier to share the indexer rather than taking addresses of arguments?

Reply:
Probably. I think I need to expose a thin API for specific indexer operations on the typechecker thread which avoids these hacky shenanigans, and then call into that from `request_dispatch` as a special case.